Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -468,10 +485,19 @@ public override void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.Event
         /// If a modal dialog is active, dispatch the pointer event to that modal dialog
         /// Returns true if the event was handled by a modal handler
         /// </summary>
-        private bool DispatchEventToObjectFocusedByPointer<T>(IMixedRealityPointer mixedRealityPointer, BaseInputEventData baseInputEventData,
+        private bool DispatchEventToObjectFocusedByPointer<T>(IMixedRealityPointer mixedRealityPointer, BaseEventData baseEventData,
             bool modalEventHandled, ExecuteEvents.EventFunction<T> eventHandler) where T : IEventSystemHandler
         {
-            GameObject focusedObject = FocusProvider?.GetFocusedObject(mixedRealityPointer);
+            GameObject focusedObject;
+            var focusEventData = baseEventData as FocusEventData;
+            if (focusEventData != null)
+            {
+                focusedObject = focusEventData.NewFocusedObject == null ? focusEventData.OldFocusedObject : focusEventData.NewFocusedObject;
```

Comment:
@luis-valverde-ms @keveleigh I just realized that this is not distinct code. If you're going to review anyways it would be nice to know if this whole thing is even the right way to do this.

Reply:
I fixed that issue.